### PR TITLE
fix: file API tests

### PR DESCRIPTION
**Pull request by Maxim-Gadalov**

### CODE REVIEW

File: src/test/java/com/epam/aidial/core/FileApiTest.java [hunk 1]
```diff
@@ -104,50 +106,64 @@ public void testFileUpload(Vertx vertx, VertxTestContext context) {
...
+                                    Assertions.assertEquals(expectedFileMetadata, response.body());
+                                    checkpoint.flag();
+                                    promise.complete();
+                                });
+                            })
+                    );
+
+            return promise.future();
+        }).compose((mapper) -> {
+            Promise<Void> promise = Promise.promise();
```
Comment (astsiapanay) [inline src/test/java/com/epam/aidial/core/FileApiTest.java]: the last promise is not required

**Comment by astsiapanay** [comment]
Please add a reference to the issue you worked on in the commit message.

File: src/main/java/com/epam/aidial/core/storage/InputStreamReader.java [hunk 2]
```diff
@@ -50,7 +50,6 @@ public InputStreamReader(Vertx vertx, InputStream in, int bufferSize) {
             }
         });
         queue.drainHandler(v -> readDataFromStream());
-        readDataFromStream();
```
Comment (artsiomkorzun) [inline src/main/java/com/epam/aidial/core/storage/InputStreamReader.java:53]: Just call queue.pause and that is it.

**Review by artsiomkorzun — APPROVED** [review]

File: src/test/java/com/epam/aidial/core/FileApiTest.java (hunk 1 above)
Comment (Maxim-Gadalov) [reply src/test/java/com/epam/aidial/core/FileApiTest.java]: fixed

**Pull request merged**